backend_tools/db_provider.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  doc,
+  Firestore,
+  getFirestore,
+  setDoc,
+  updateDoc,
+  addDoc,
+  getDoc,
+  DocumentData,
+  Query,
+  where,
+  getDocs,
+  QueryDocumentSnapshot,
+  query,
+  QuerySnapshot,
+  deleteDoc,
+  collection,
+  arrayUnion,
+} from 'firebase/firestore';
+import Club from '../models/club';
+import User from '../models/user';
+import FirebaseApp from './firebase';
+import Event from '../models/event';
+
+class DbProvider {
+  db: Firestore = getFirestore(FirebaseApp);
+  user_path: string = 'Users';
+  club_path: string = 'Clubs';
+  event_path: string = 'Events';
+  async createUser(user: User): Promise<boolean> {
+    try {
+      const userCollection = collection(this.db, this.user_path);
+      await addDoc(userCollection, user);
+      return true;
+    } catch (error) {
+      console.error(`failure saving user, got error: ${error}`);
+      return false;
+    }
+  }
+
+  async createClub(club: Club): Promise<boolean> {
```

Comment:
Maybe instead of returning true or false return the club id | undefined

Reply:
Same thing w/ user

Reply:
The club id for this function is the name of the club